diff --git a/gi/meson.build b/gi/meson.build
index c1afd68..15a6521 100644
--- a/gi/meson.build
+++ b/gi/meson.build
@@ -76,6 +76,7 @@ giext = python.extension_module('_gi', sources,
   dependencies : [python_ext_dep, glib_dep, gi_dep, ffi_dep],
   include_directories: include_directories('..'),
   install: true,
+  install_rpath: join_paths(get_option('prefix'), get_option('libdir')),
   subdir : 'gi',
   c_args: pyext_c_args + main_c_args
 )
@@ -84,6 +85,7 @@ if with_pycairo
   gicairoext = python.extension_module('_gi_cairo', ['pygi-foreign-cairo.c'],
     dependencies : [python_ext_dep, glib_dep, gi_dep, ffi_dep, pycairo_dep, cairo_dep, cairo_gobject_dep],
     install: true,
+    install_rpath: join_paths(get_option('prefix'), get_option('libdir')),
     subdir : 'gi',
     c_args: pyext_c_args + main_c_args)
 endif
